File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        
        int l=0, r=numbers.size()-1;
        vector<int> op;

        while(l <= r)
        {
            if(numbers[l] + numbers[r] == target)
            {
                op.push_back(l+1);
                op.push_back(r+1);
                return op;
            }
            else if(numbers[l] + numbers[r] < target)
            {
                l++;
            }
            else
            {
                r--;
            }
        }

        return op;
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        l = 0
        r = len(numbers)

        dict = {}

        for i in range(len(numbers)):
            bal = target - numbers[i]
            
            if bal in dict:
                return [dict[bal], i+1]
            
            dict[numbers[i]] = dict.get(numbers[i], i)+1

        return [-1,-1]